### ISSUE TITLE
Travis build failing on PHP 5.4

### ISSUE THREAD

**Issue by pfrenssen** [jhedstrom/drupalextension#186]
Since Drupal 8 is now requiring PHP 5.5, the Travis build fails for PHP 5.4:

```
Your PHP installation is too old. Drupal requires at least PHP <em class="placeholder">5.5.9</em>.
```

I did some experimentation the other day and we cannot run the tests on PHP 5.6 or PHP 7 because it is impossible to install Drupal 6 on those. This is most likely due to a configuration problem on the Travis test servers since I was able to install Drupal 6 without problems on my local dev machine with PHP 5.6

So we are currently in the awkward position that the tests only pass on a very narrow band of available versions: PHP 5.5 or naught :)

Would a good solution be to start using a full build matrix that captures the allowed versions for each? Something like this perhaps:

```
php:
  - 5.4
  - 5.5
  - 5.6
  - 7.0
env:
  - DRUPAL_VERSION=6
  - DRUPAL_VERSION=7
  - DRUPAL_VERSION=8

matrix:
  exclude:
    - php: 5.4
      env: DRUPAL_VERSION=8
    - php: 5.6
      env: DRUPAL_VERSION=6
    - php: 7.0
      env: DRUPAL_VERSION=6
```

**Comment by jhedstrom** [comment]
:+1: for a build matrix. We could then add php 5.3 back for Drupal 6 and 7.

**Comment by pfrenssen** [comment]
I'm experimenting with a build matrix in pull request #187.

**Comment by pfrenssen** [comment]
I'm making some progress, seeing some green builds appearing in the pull request after some experimentation. It does seem like testing PHP 5.3 is out of the question. Both Drush and the PHP built-in webserver require PHP 5.4 to run.

We could work around both by using an older Drush version and an Apache installation when running a PHP 5.3 test but that seems to be a lot of trouble for testing an unsupported PHP version.

**Comment by pfrenssen** [comment]
I finished the work on the pull request. I had initially added support for HHVM but removed it again since the test machines became unresponsive on the 'echo sendmail_path' step and were killed by the Travis supervisor after 10 minutes.

Once I got a green build I have deliberately introduced a test failure to verify that the failures are still correctly detected. All seems to be well.

Tests are currently still in progress but they should normally be green.
